packages/wdio-cli/src/cjs/index.ts

```diff
@@ -0,0 +1,115 @@
+import type { RunCommandArguments } from '../types.js'
+import type { Options, Capabilities } from '@wdio/types'
+import type { EndMessage } from '../launcher.js'
+
+export class Launcher {
+
+    esmLauncher: any
+
+    constructor(
+        private configFilePath: string,
+        private args: Partial<RunCommandArguments> = {},
+        private isWatchMode = false
+    ) {
+        import('../index.js').then(esmModule => this.esmLauncher = new esmModule.Launcher(this.configFilePath, this.args, this.isWatchMode))
+    }
+
+    /**
+     * run sequence
+     * @return  {Promise}  that only gets resolves with either an exitCode or an error
+     */
+    async run() {
+        return this.esmLauncher.run()
+    }
+
+    /**
+     * run without triggering onPrepare/onComplete hooks
+     */
+    runMode (config: Required<Options.Testrunner>, caps: Capabilities.RemoteCapabilities): Promise<number> {
+        return this.esmLauncher.runMode(config, caps)
+    }
+
+    /**
+     * Format the specs into an array of objects with files and retries
+     */
+    formatSpecs(capabilities: (Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities | Capabilities.RemoteCapabilities), specFileRetries: number) {
+        this.esmLauncher.formatSpecs(capabilities, specFileRetries)
+    }
+
+    /**
+     * run multiple single remote tests
+     * @return {Boolean} true if all specs have been run and all instances have finished
+     */
+    runSpecs() {
+        return this.esmLauncher.runSpecs()
+    }
+
+    /**
+     * gets number of all running instances
+     * @return {number} number of running instances
+     */
+    getNumberOfRunningInstances() {
+        return this.esmLauncher.getNumberOfRunningInstances()
+    }
+
+    /**
+     * get number of total specs left to complete whole suites
+     * @return {number} specs left to complete suite
+     */
+    getNumberOfSpecsLeft() {
+        return this.esmLauncher.getNumberOfSpecsLeft()
+    }
+
+    /**
+     * Start instance in a child process.
+     * @param  {Array} specs  Specs to run
+     * @param  {Number} cid  Capabilities ID
+     * @param  {String} rid  Runner ID override
+     * @param  {Number} retries  Number of retries remaining
+     */
+    async startInstance(
+        specs: string[],
+        caps: Capabilities.DesiredCapabilities | Capabilities.W3CCapabilities | Capabilities.MultiRemoteCapabilities,
+        cid: number,
+        rid: string | undefined,
+        retries: number
+    ) {
+        this.esmLauncher.startInstance(specs, caps, cid, rid, retries)
+    }
+
+    /**
+     * generates a runner id
+     * @param  {Number} cid capability id (unique identifier for a capability)
+     * @return {String}     runner id (combination of cid and test id e.g. 0a, 0b, 1a, 1b ...)
+     */
+    getRunnerId (cid: number) {
```

Comment:
can we make a bunch of these methods private so we don't have to expose them here? The launcher has been build before private class functions in JavaScript were a thing. Maybe we can optimize the code here so we only have to expose the `run` method which is the only one we've documented.

Reply:
Some of the methods I refactored to private methods. Not sure how many people actually use this, so kept a few methods public for now.
Also happy making most of them private.

Reply:
> Also happy making most of them private.

Let's make all of them private except `run`. We don't have anything else documented and therefor people should not use any other method. If folks start to report this, we can think about making them public again.

Reply:
Yeah you are right. Just did that.